src/index.js

```diff
@@ -47,34 +47,48 @@ app.post('/interactive-message', (req, res) => {
   const payload = JSON.parse(req.body.payload);
   const token = payload.token;
   const actions = payload.actions;
-  const callback_id = payload.callback_id;
+  const action_id = actions[0].action_id;
+  const action_array = action_id.split(".");
+  var fieldName = action_array[0];
+  const ticket_id = action_array[1];
   const response_url = payload.response_url;
   const user = payload.user;
 
   if (token === process.env.SLACK_VERIFICATION_TOKEN) {
     // Immediately respond to signal success, further updates will be done using `response_url`
     res.send('');
-
-    Ticket.find(callback_id).then((ticket) => {
+    Ticket.find(ticket_id).then((ticket) => {
       debug('interactive message ticket found');
       const action = actions[0];
-      const fieldName = action.name;
-      // Handle either message menu action or message button (Claim)
-      const fieldValue = action.selected_options ? action.selected_options[0].value : user.id;
+      switch (fieldName) {
+        case 'claim':
+          fieldName = 'agent';
+          var fieldValue = user.id;
+          break;
+        case 'agent':
+          var fieldValue = action.selected_user;
+          break;
+        case 'priority':
+          var fieldValue = action.selected_option.value;
+          break;
+
+        default:
+          break;
+      }
 
       return ticket.updateField(fieldName, fieldValue).then(() => {
         debug('updating notification in channel');
         return ticket.postToChannel(response_url);
       });
     })
-    // Error handling
-    .catch(console.error);
+      // Error handling
+      .catch(console.error);
   } else {
     debug('check verification token failed');
     res.sendStatus(404);
   }
 });
 
 app.listen(process.env.PORT, () => {
-  console.log(`App listening on port ${process.env.PORT}!`);
+  debug(`App listening on port ${process.env.PORT}!`);
```

Comment:
maybe not convert this to `debug()`. the idea is that the developer always wants to see this output on stdout, which `debug` stuff is to get detailed debugging info when certain environment variables are set. you kind of *have* to see this message to know your app started up correctly.

Reply:
🤦‍♂️ Find and Replace fail...